metrics/influxdb.go

```diff
@@ -19,26 +19,22 @@ type InfluxDB struct {
 	database string
 }
 
-// InitializeClient created a new UDP based influx db client. The more secure way of doing would be to use http client
-// with batching to avoid lock-ups. For now, UDP client is easier to implement with the useful non-blocking property.
-// TODO: use HTTP client with batching instead of UDP.
+// InitializeClient creates a new HTTP based InfluxDB client. This client will be used for the lifetime of the application.
 func (flxDB *InfluxDB) InitializeClient(addr string, port int, database string) error {
 	var err error
-	flxDB.client, err = influxdb_client.NewUDPClient(influxdb_client.UDPConfig{
-		Addr:        fmt.Sprintf("%s:%d", addr, port),
-		PayloadSize: 512,
+	flxDB.client, err = influxdb_client.NewHTTPClient(influxdb_client.HTTPConfig{
+		Addr: fmt.Sprintf("http://%s:%d", addr, port),
 	})
-	flxDB.port = port
-	flxDB.addr = addr
-	flxDB.database = database
 	if err != nil {
 		return err
 	}
+	flxDB.port = port
+	flxDB.addr = addr
+	flxDB.database = database
 	return nil
 }
 
-// EmitSingle sends a single metric out using the current influx db client. It implicitly adds the source host where
-// the metric is coming from. This operation is non-blocking since we use the UDP client.
+// EmitSingle sends a single metric out using the current InfluxDB client. It adds the source host where the metric is coming from.
 func (flxDB *InfluxDB) EmitSingle(m SingleMetric) {
```

Comment:
Batching implementation here seems misplaced. We call emitSingle in main.go:71 in a loop and with this change we will still make multiple calls to flxDB.client.Write(). We want to avoid calling it multiple times.

 I'd keep this method as is, because we might need it in future. Instead, I'd create a new method called something like: collectMetrics or similar. And another method called: emitMultiple or something like that.  And then in main.go:71 we can collect metrics and in the end of the iteration we call emitMultiple - which will issue just one send.

What do you think ?

Currently, we emit a single metric from a loop in main.go:71. We should probably batch the metrics from there and then send it in one go.

Reply:
Yep this will help to avoid multiple writes to the database. I think we can add it in a new branch.

Reply:
If you plan to add proper batching in another branch, then I'd suggest not changing this method now. Let's have another pull request dedicated for batching where we can discuss options.

Reply:
Added CollectMetrics and EmitMultiple methods for batching and sending metrics in bulk to the database, reducing write operations and improving performance. The reason i decided to discuss and implement this in this pull request is that it turned out to be easier than i thought.